### ISSUE TITLE
Threshold Overfitting

### ISSUE THREAD

**Issue by Mannava-Daasaradhi** [PrajithS20/ST-QAN-ViT#2]
The find_balanced_threshold function scans the test data to find the optimal threshold that gives the best results.
This is tuning hyperparameters on the test set. The reported performance will be optimistic and likely won't hold up on new, unseen data.
Split the data into Train/Val/Test. Find the threshold on Val, then use that fixed threshold to report metrics on Test.

**Comment by PrajithS20** [comment]
We have implemented a strict validation separation in 03_train_model.py which exports a frozen test_files_manifest.txt. This allows the optimal threshold to be tuned strictly on the validation set before being applied to unseen test data for unbiased reporting.
